library/Requests/Transport/fsockopen.php

```diff
@@ -121,62 +121,68 @@ public function request($url, $headers = array(), $data = array(), $options = ar
 				// Connection issue
 				throw new Requests_Exception(rtrim($this->connect_error), 'fsockopen.connect_error');
 			}
+
+			throw new Requests_Exception($errstr, 'fsockopenerror');
+		}
+
+		$data_format = $options['data_format'];
+
+		if ($data_format === 'default') {
+			if (in_array($options['type'], array(Requests::HEAD, Requests::GET, Requests::DELETE))) {
+				$data_format = 'query';
+			}
 			else {
-				throw new Requests_Exception($errstr, 'fsockopenerror');
-				return;
+				$data_format = 'body';
 			}
 		}
 
+		if ($data_format === 'query') {
+			$path = self::format_get($url_parts, $data);
+			$data = '';
+		}
+		else {
+			$path = self::format_get($url_parts, array());
+		}
+
+		$options['hooks']->dispatch('fsockopen.remote_host_path', array(&$path, $url));
```

Comment:
Not sure about this one. Should I only dispatch it for the `query` data format in order to make it fully BC?